File: src/plugin/item/AncientEffigyItemPlugin.java
package plugin.item;

import org.redrune.game.content.plugin.type.ItemPlugin;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.entity.item.Item;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 2019-02-08
 */
public class AncientEffigyItemPlugin implements ItemPlugin {
	
	@Override
	public boolean handle(Player player, Item item, int slotId, String option) {
		player.getDialogueManager().startDialogue("AncientEffigiesD", item.getId());
		return true;
	}
	
	@Override
	public void register() {
		for (int i = 18788; i <= 18781; i++) {
			registerItem(i, "Investigate");
		}
	}
}

File: src/main/java/plugin/item/AncientEffigyItemPlugin.kt
package plugin.item

import game.content.plugin.type.ItemPlugin
import game.entity.actor.player.Player
import game.entity.item.Item

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-02-08
 */
class AncientEffigyItemPlugin : ItemPlugin {
    override fun handle(player: Player, item: Item, slotId: Int, option: String): Boolean {
        player.dialogueManager.startDialogue("AncientEffigiesD", item.id)
        return true
    }

    override fun register() {
        for (i in 18788 downTo 18781) {
            registerItem(i, "Investigate")
        }
    }
}